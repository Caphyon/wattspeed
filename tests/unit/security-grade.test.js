import { createLocalVue, shallowMount } from '@vue/test-utils';
import security_grade from '../../src/components/panels/tech/items/security-grade.vue';

const localVue = createLocalVue();

const data = {
  grade: {
    grade: '',
    colour: '',
  },
  section: '',
};

const noData = {};

function createWrapper(data) {
  return shallowMount(security_grade, {
    localVue,
    propsData: data,
  });
}

describe('Test security grade', () => {
  test('Render security-grade with no data', () => {
    const wrapper = createWrapper(noData);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--secondary').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe('NA');
    expect(wrapper.find('.security-grade-section').text()).toBe('');
    wrapper.destroy();
  });

  test('Render grade A+ with color green', () => {
    data.grade.grade = 'A+';
    data.grade.colour = 'green';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--success').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade A with color green', () => {
    data.grade.grade = 'A';
    data.grade.colour = 'green';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--success').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade B with color yellow', () => {
    data.grade.grade = 'B';
    data.grade.colour = 'yellow';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--warning').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade C with color yellow', () => {
    data.grade.grade = 'C';
    data.grade.colour = 'yellow';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--warning').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade D with color orange', () => {
    data.grade.grade = 'D';
    data.grade.colour = 'orange';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--orange').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade E with color orange', () => {
    data.grade.grade = 'E';
    data.grade.colour = 'orange';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--orange').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade F with color red', () => {
    data.grade.grade = 'F';
    data.grade.colour = 'red';
    data.section = 'Headers';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--danger').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

  test('Render grade T with color red', () => {
    data.grade.grade = 'T';
    data.grade.colour = 'red';
    data.section = 'SSL';
    const wrapper = createWrapper(data);

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.security-grade-letter.alert--danger').exists()).toBeTruthy();
    expect(wrapper.find('.security-grade-letter').text()).toBe(data.grade.grade);
    expect(wrapper.find('.security-grade-section').text()).toBe(data.section);
    wrapper.destroy();
  });

});