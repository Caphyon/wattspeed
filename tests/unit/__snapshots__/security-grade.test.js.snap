// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test security grade Render grade A with color green 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--success"
  >
    A
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade A+ with color green 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--success"
  >
    A+
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade B with color yellow 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--warning"
  >
    B
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade C with color yellow 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--warning"
  >
    C
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade D with color orange 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--orange"
  >
    D
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade E with color orange 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--orange"
  >
    E
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade F with color red 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--danger"
  >
    F
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    Headers
  </span>
</div>
`;

exports[`Test security grade Render grade T with color red 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--danger"
  >
    T
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    SSL
  </span>
</div>
`;

exports[`Test security grade Render security-grade with no data 1`] = `
<div
  class="text--center"
>
  <h3
    class="security-grade-letter mt0 mb0 alert--secondary"
  >
    NA
  </h3>
   
  <span
    class="security-grade-section text--uppercase text--small"
  >
    
  </span>
</div>
`;
